From b42744b629aa721b88b9522e507f612d5198caff Mon Sep 17 00:00:00 2001
From: kieth liu <kieth.liu@amlogic.com>
Date: Thu, 10 Jun 2021 21:34:04 +0800
Subject: [PATCH] PD#SWPL-52003

Problem:
many tv Provider onChange for dvb
player that time consuming

Solution:
rm every Uri change need target notifyChange
in applyBatch, need notifyChange when all Uri
update finished

Verify:
ok on redi.

Change-Id: Iddbb2ff5ddcb574064f53bc8a122016ef39c5bb3
Signed-off-by: kieth liu <kieth.liu@amlogic.com>
---
 src/com/android/providers/tv/TvProvider.java | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/src/com/android/providers/tv/TvProvider.java b/src/com/android/providers/tv/TvProvider.java
index d49d387..7e6d95f 100644
--- a/src/com/android/providers/tv/TvProvider.java
+++ b/src/com/android/providers/tv/TvProvider.java
@@ -1994,11 +1994,7 @@ public class TvProvider extends ContentProvider {
             return results;
         } finally {
             db.endTransaction();
-            final Set<Uri> notifications = getBatchNotificationsSet();
             setBatchNotificationsSet(null);
-            for (final Uri uri : notifications) {
-                context.getContentResolver().notifyChange(uri, null);
-            }
         }
     }
 
-- 
2.29.0

